client/router.js

```diff
@@ -36,12 +37,20 @@ Router.prototype.route = function(path, options) {
   };
 
   this._routes.push(route);
+  if (options.name) {
+    this._pathsMap[options.name] = path;
+  }
+
   this._updateCallbacks();
 
   return route;
 };
 
 Router.prototype.path = function(pathDef, fields, queryParams) {
+  if (this._pathsMap[pathDef]) {
```

Comment:
Just thinking. Do we need to ever worry about a clash between route name and a path.